### ISSUE TITLE
Short array syntax not allowed drives me nuts

### ISSUE THREAD

**Issue by Micu22** [WordPress/WordPress-Coding-Standards#1968]
**I ask for a change to the standard and a simple setting to disable the error „Short array syntax is not allowed” in the meantime.**

## Solution:
I want to write the more readable and easier to understand for beginners square brackets [] to define arrays.

1. They are more readable, because the closing parenthesis "]" has meaning, while ")" means nothing, and one has to look for longer in the code.
2. They are easier on beginners because they are consistent. You access array data the same way you create it: `$myarray = [ "dog" => "barks", "cat" => "meows" ];` and then `$myarray[ "dog" ];`.

 And until it's changed I ask for someone to tell me how can I disable this one setting, because red dots in Sublime Text appear next to every single array declaration and it drives me and my co-workers crazy.

One last word. I am writing here only because I actually care about the coding standard. I could disable wpcs entirely.

**Comment by SumoTTo** [comment]
Hello!

You can try disabling the warnings by following the instructions on this page. https://github.com/WordPress/WordPress-Coding-Standards/wiki/Whitelisting-code-which-flags-errors

In your case, you can disable errors for each specific line
```
$myarray = [ 'dog' => 'barks', 'cat' => 'meows' ]; // phpcs:ignore
$myarray = [ 'dog' => 'barks', 'cat' => 'meows' ]; // phpcs:ignore WordPress.Arrays.ArrayDeclarationSpacing, Generic.Arrays.DisallowShortArraySyntax
```
or, which is probably more correct, use the settings file https://github.com/WordPress/WordPress-Coding-Standards#using-a-custom-ruleset and indicate in it
```
<rule ref="WordPress-Extra">
	<exclude name="WordPress.Arrays.ArrayDeclarationSpacing"/>
	<exclude name="Generic.Arrays.DisallowShortArraySyntax"/>
</rule>
```
I don't have Sublime Text at hand to test it in, but it definitely works in PhpStorm.

**Comment by jrfnl** [comment]
> I ask for a change to the standard

Your opinion is noted, but as opinions on short vs long arrays are deeply divided, the standard is unlikely to chance regarding this any time soon.

@dingo-d Already pointed you to the earlier discussions on this. Please add your voice there.

> a simple setting to disable the error „Short array syntax is not allowed” in the meantime

That has always been available as it's a default PHPCS feature - any individual sniff or error message can be turned on or off.

On the command line:
```bash
phpcs ./folder-to-scan/ --standard=WordPress --exclude=Generic.Arrays.DisallowShortArraySyntax
```

Or in a custom ruleset:
```xml
<rule ref="WordPress">
    <exclude name="Generic.Arrays.DisallowShortArraySyntax"/>
</rule>

<!-- Optionally enforce short arrays: -->
<rule ref="Generic.Arrays.DisallowLongArraySyntax"/>
```

Refs:
* https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#limiting-results-to-specific-sniffs
* https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-Ruleset

**Comment by Micu22** [comment]
Ok, not sure where should I leave my voice, I don't see any mention about this in my previous discussion.

I got it sorted, previously I was trying 
`<exclude name="WordPress.Arrays"/>`
and it didn't work, now this works flawlessly:
`<exclude name="Generic.Arrays.DisallowShortArraySyntax"/>`

Also if someone is struggling with this in Sublime, you have to remove this from Linter settings:
```
"args": [
     "--standard=WordPress-Extra"
]
```
because it will strangely not look at the custom ruleset if there is a top level standard defined

**Comment by sandrodz** [comment]
I agree with @Micu22 

I also noticed this `Generic.Arrays.DisallowShortArraySyntax.Found` just now, I was using old versions of wpcs and it didn't have this rule enforce.

<img width="370" alt="Screen Shot 2021-12-19 at 22 43 04" src="https://user-images.githubusercontent.com/8479569/146686966-16180d4c-73e2-42b2-8ed0-a53388347c8b.png">

**Comment by sherif-magdy** [comment]
Adding to @jrfnl, I got to exclude 'Universal.Arrays.DisallowShortArraySyntax' sniff to get rid of that error: 

```html
<rule ref="WordPress">
    <exclude name="Generic.Arrays.DisallowShortArraySyntax"/>
    <exclude name="Universal.Arrays.DisallowShortArraySyntax"/>
</rule>

<!-- Optionally enforce short arrays: -->
<rule ref="Generic.Arrays.DisallowLongArraySyntax"/>
```

**Comment by sybrew** [comment]
>  but as opinions on short vs long arrays are deeply divided

Do you know where I can read up on this? All I've seen is a sweeping preference for the short array syntax.

**Comment by dingo-d** [comment]
> > but as opinions on short vs long arrays are deeply divided
> 
> Do you know where I can read up on this? All I've seen is a sweeping preference for the short array syntax.

https://make.wordpress.org/core/2019/07/12/php-coding-standards-changes/

This was the proposal where it was shut down. You can also search this repo and check the history.

**Comment by sybrew** [comment]
I don't see anyone supporting the long array syntax or offering a reasonable explanation of the topics raised here.

Please stop the gaslighting. Thanks.

**Comment by jrfnl** [comment]
> Please stop the gaslighting. Thanks.

@sybrew Please refrain from attacking people who are just pointing you to the answer to your question. Consider this a CoC warning.

**Comment by sybrew** [comment]
The answer he pointed to contrasted with what he was saying.

![Image](https://github.com/user-attachments/assets/120b9d2a-eb9d-4496-9ff3-6739404a6116)

![Image](https://github.com/user-attachments/assets/0a2d54ac-5754-4cb9-9671-8c29038d586a)

![Image](https://github.com/user-attachments/assets/e3b31dc9-f166-41c1-a72e-ef54890b1733)

![Image](https://github.com/user-attachments/assets/16fe77a4-c575-4a5c-8f2f-9740cdcc4c38)

![Image](https://github.com/user-attachments/assets/a6ff657b-7a8e-4f6a-9848-69ceb12df063)

![Image](https://github.com/user-attachments/assets/1f805fd0-c623-4991-8471-9561088a128e)

![Image](https://github.com/user-attachments/assets/2bacbdb3-0710-46d1-83e9-3799f0a69fdc)

![Image](https://github.com/user-attachments/assets/c580f6d8-ade5-4e0f-9c6f-1e5e6575ad88)

**Comment by jrfnl** [comment]
@sybrew Pasting a dozen screenshots is not helpful (and not welcome). Everyone can read the comments in the original threads themselves, just like they can read the previous tickets. Please refrain from this type of disruptive behaviour.

The short of it is simple: there is overwhelming support from the wider WP community for allowing short arrays, but amongst the committers, the opinions are far more divided, including a lot of trepidation about the impact of such a change on open patches.

It doesn't matter whether you or I agree with that sentiment. In the end, if there is no consensus or wide support from the committers, that's the deciding factor.

**Comment by sybrew** [comment]
I appreciate the clarification you gave. Thank you.

It is concerning, however, that the committers' preferences (however much they're escaping public discussions and consensus) are trickling their way into this project, and then the Plugin Check plugin, with which they're now considering [imposing these guidelines as rules](https://make.wordpress.org/plugins/2024/12/24/plugin-check-goals-roadmap/). This impactful change makes creating and upholding a public discourse over these decisions critical. I'm now trying a low-hanging fruit, but seeing how even that is barred, I'm afraid for the future.

I'll leave it at that. Thank you.

**Comment by jrfnl** [comment]
@sybrew In my opinion, Plugin Check should not have any rules related to code _style_ only to code **quality**, but that's something to discuss with the Plugin team and not here.
